### ISSUE TITLE
Nested Schemas not Rendering Properly

### ISSUE THREAD

**Issue by tahoward** [swagger-api/swagger-ui#5194]
### Q&A (please complete the following information)
 - OS: macOS
 - Browser: Chrome
 - Method of installation: `pip install swagger-ui-bundle`
 - Swagger-UI version: 3.20.5
 - Swagger/OpenAPI version: OpenAPI 3.0

### Content & configuration
<details><summary>Broken</summary>
<p>

```js
{
  "openapi": "3.0.0",
  "info": {
    "title": "API",
    "version": "1.0"
  },
  "servers": [{
    "url": "/1.0"
  }],
  "components": {
    "schemas": {
      "BaseError": {
        "required": [
          "code",
          "message"
        ],
        "type": "object",
        "properties": {
          "message": {
            "type": "string"
          },
          "code": {
            "type": "integer"
          }
        }
      },
      "BaseStringList": {
        "type": "array",
        "items": {
          "type": "string"
        }
      },
      "BaseLink": {
        "type": "object",
        "properties": {
          "rel": {
            "type": "string"
          },
          "href": {
            "type": "string",
            "format": "uri"
          }
        }
      },
      "BaseLinkList": {
        "type": "object",
        "properties": {
          "links": {
            "type": "array",
            "items": {
              "$ref": "#/components/schemas/BaseLink"
            }
          }
        }
      },
      "BaseTeam": {
        "type": "object",
        "properties": {
          "teamName": {
            "type": "string"
          }
        }
      },
      "LinkedTeam": {
        "allOf": [{
          "$ref": "#/components/schemas/BaseTeam"
        }, {
          "$ref": "#/components/schemas/BaseLinkList"
        }]
      },
      "BaseTeamList": {
        "type": "object",
        "properties": {
          "data": {
            "type": "array",
            "items": {
              "$ref": "#/components/schemas/LinkedTeam"
            }
          }
        }
      },
      "LinkedTeamList": {
        "allOf": [{
          "$ref": "#/components/schemas/BaseTeamList"
        }, {
          "$ref": "#/components/schemas/BaseLinkList"
        }]
      },
      "BaseADGroups": {
        "type": "object",
        "properties": {
          "deploy": {
            "$ref": "#/components/schemas/BaseStringList"
          }
        }
      },
      "BaseVPC": {
        "type": "object",
        "properties": {
          "cidr": {
            "type": "string"
          }
        }
      },
      "BaseSubnet": {
        "type": "object",
        "properties": {
          "vpc": {
            "$ref": "#/components/schemas/BaseVPC"
          }
        }
      },
      "BaseRegion": {
        "type": "object",
        "properties": {
          "regionName": {
            "type": "string"
          },
          "subnet": {
            "$ref": "#/components/schemas/BaseSubnet"
          }
        }
      },
      "BaseRegionList": {
        "type": "array",
        "items": {
          "$ref": "#/components/schemas/BaseRegion"
        }
      },
      "BaseAccountModifiable": {
        "type": "object",
        "properties": {
          "adGroups": {
            "$ref": "#/components/schemas/BaseADGroups"
          }
        }
      },
      "BaseAccountCreate": {
        "allOf": [{
          "$ref": "#/components/schemas/BaseAccountModifiable"
        }, {
          "type": "object",
          "properties": {
            "organizationName": {
              "type": "string"
            },
            "teamName": {
              "maxLength": 12,
              "minLength": 1,
              "type": "string"
            }
          }
        }]
      },
      "BaseAccount": {
        "allOf": [{
          "$ref": "#/components/schemas/BaseAccountCreate"
        }, {
          "type": "object",
          "properties": {
            "regionMapping": {
              "$ref": "#/components/schemas/BaseRegionList"
            },
            "accountName": {
              "type": "string"
            },
            "accountStatus": {
              "type": "string"
            }
          }
        }]
      },
      "LinkedAccount": {
        "allOf": [{
          "$ref": "#/components/schemas/BaseAccount"
        }, {
          "$ref": "#/components/schemas/BaseLinkList"
        }]
      },
      "BaseAccountList": {
        "type": "object",
        "properties": {
          "data": {
            "type": "array",
            "items": {
              "$ref": "#/components/schemas/LinkedAccount"
            }
          }
        }
      },
      "LinkedAccountList": {
        "allOf": [{
          "$ref": "#/components/schemas/BaseAccountList"
        }, {
          "$ref": "#/components/schemas/BaseLinkList"
        }]
      }
    },
    "responses": {
      "NotFound": {
        "description": "Resource was not found",
        "content": {
          "application/json": {
            "schema": {
              "$ref": "#/components/schemas/BaseError"
            }
          }
        }
      },
      "Root": {
        "description": "List of Routes",
        "content": {
          "application/json": {
            "schema": {
              "$ref": "#/components/schemas/BaseLinkList"
            }
          }
        }
      },
      "TeamList": {
        "description": "List of Teams",
        "content": {
          "application/json": {
            "schema": {
              "$ref": "#/components/schemas/LinkedTeamList"
            }
          }
        }
      },
      "AccountList": {
        "description": "List of Accounts",
        "content": {
          "application/json": {
            "schema": {
              "$ref": "#/components/schemas/LinkedAccountList"
            }
          }
        }
      },
      "Account": {
        "description": "An Account",
        "content": {
          "application/json": {
            "schema": {
              "$ref": "#/components/schemas/LinkedAccount"
            }
          }
        }
      }
    },
    "parameters": {
      "TeamName": {
        "name": "teamName",
        "in": "path",
        "description": "The team name.",
        "required": true,
        "style": "simple",
        "explode": false,
        "schema": {
          "type": "string"
        }
      },
      "Environment": {
        "name": "environment",
        "in": "path",
        "description": "The account environment.",
        "required": true,
        "style": "simple",
        "explode": false,
        "schema": {
          "type": "string"
        }
      }
    },
    "requestBodies": {
      "AccountCreate": {
        "description": "Account Update JSON",
        "content": {
          "application/json": {
            "schema": {
              "$ref": "#/components/schemas/BaseAccountCreate"
            }
          }
        },
        "required": true
      },
      "AccountModify": {
        "description": "Account Create JSON",
        "content": {
          "application/json": {
            "schema": {
              "$ref": "#/components/schemas/BaseAccountModifiable"
            }
          }
        },
        "required": true
      }
    }
  },
  "paths": {
    "/": {
      "get": {
        "summary": "List All Routes",
        "operationId": "api.get_list",
        "responses": {
          "200": {
            "description": "List of Routes",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/BaseLinkList"
                }
              }
            }
          }
        }
      }
    },
    "/accounts": {
      "get": {
        "summary": "List All Accounts",
        "operationId": "api.account.get_list",
        "responses": {
          "200": {
            "description": "List of Accounts",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/LinkedAccountList"
                }
              }
            }
          }
        }
      },
      "post": {
        "summary": "Create an Account",
        "operationId": "api.account.create",
        "requestBody": {
          "$ref": "#/components/requestBodies/AccountCreate"
        },
        "responses": {
          "200": {
            "description": "An Account",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/LinkedAccount"
                }
              }
            }
          }
        }
      }
    },
    "/teams": {
      "get": {
        "summary": "List All Teams",
        "operationId": "api.team.get_list",
        "responses": {
          "200": {
            "description": "List of Teams",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/LinkedTeamList"
                }
              }
            }
          }
        }
      }
    },
    "/teams/{teamName}": {
      "get": {
        "summary": "List Team Accounts",
        "operationId": "api.team.get",
        "parameters": [{
          "name": "teamName",
          "in": "path",
          "description": "The team name.",
          "required": true,
          "style": "simple",
          "explode": false,
          "schema": {
            "type": "string"
          }
        }],
        "responses": {
          "200": {
            "description": "List of Accounts",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/LinkedAccountList"
                }
              }
            }
          },
          "404": {
            "description": "Resource was not found",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/BaseError"
                }
              }
            }
          }
        }
      }
    },
    "/teams/{teamName}/{environment}": {
      "get": {
        "summary": "Get Team Account",
        "operationId": "api.account.get",
        "parameters": [{
          "name": "teamName",
          "in": "path",
          "description": "The team name.",
          "required": true,
          "style": "simple",
          "explode": false,
          "schema": {
            "type": "string"
          }
        }, {
          "name": "environment",
          "in": "path",
          "description": "The account environment.",
          "required": true,
          "style": "simple",
          "explode": false,
          "schema": {
            "type": "string"
          }
        }],
        "responses": {
          "200": {
            "description": "An Account",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/LinkedAccount"
                }
              }
            }
          },
          "404": {
            "description": "Resource was not found",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/BaseError"
                }
              }
            }
          }
        }
      },
      "patch": {
        "summary": "Update Team Account",
        "operationId": "api.account.update",
        "parameters": [{
          "name": "teamName",
          "in": "path",
          "description": "The team name.",
          "required": true,
          "style": "simple",
          "explode": false,
          "schema": {
            "type": "string"
          }
        }, {
          "name": "environment",
          "in": "path",
          "description": "The account environment.",
          "required": true,
          "style": "simple",
          "explode": false,
          "schema": {
            "type": "string"
          }
        }],
        "requestBody": {
          "$ref": "#/components/requestBodies/AccountModify"
        },
        "responses": {
          "200": {
            "description": "An Account",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/LinkedAccount"
                }
              }
            }
          },
          "404": {
            "description": "Resource was not found",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/BaseError"
                }
              }
            }
          }
        }
      }
    }
  }
}
```

</p>
</details>

<details><summary>Working</summary>
<p>

```js
{
  "openapi": "3.0.0",
  "info": {
    "title": "API",
    "version": "1.0"
  },
  "servers": [{
    "url": "/1.0"
  }],
  "paths": {
    "/": {
      "get": {
        "summary": "List All Routes",
        "operationId": "api.get_list",
        "responses": {
          "200": {
            "description": "List of Routes",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/BaseLinkList"
                }
              }
            }
          }
        }
      }
    },
    "/accounts": {
      "get": {
        "summary": "List All Accounts",
        "operationId": "api.account.get_list",
        "responses": {
          "200": {
            "description": "List of Accounts",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/LinkedAccountList"
                }
              }
            }
          }
        }
      },
      "post": {
        "summary": "Create an Account",
        "operationId": "api.account.create",
        "requestBody": {
          "$ref": "#/components/requestBodies/AccountCreate"
        },
        "responses": {
          "200": {
            "description": "An Account",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/LinkedAccount"
                }
              }
            }
          }
        }
      }
    },
    "/teams": {
      "get": {
        "summary": "List All Teams",
        "operationId": "api.team.get_list",
        "responses": {
          "200": {
            "description": "List of Teams",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/LinkedTeamList"
                }
              }
            }
          }
        }
      }
    },
    "/teams/{teamName}": {
      "get": {
        "summary": "List Team Accounts",
        "operationId": "api.team.get",
        "parameters": [{
          "name": "teamName",
          "in": "path",
          "description": "The team name.",
          "required": true,
          "style": "simple",
          "explode": false,
          "schema": {
            "type": "string"
          }
        }],
        "responses": {
          "200": {
            "description": "List of Accounts",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/LinkedAccountList"
                }
              }
            }
          },
          "404": {
            "description": "Resource was not found",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/BaseError"
                }
              }
            }
          }
        }
      }
    },
    "/teams/{teamName}/{environment}": {
      "get": {
        "summary": "Get Team Account",
        "operationId": "api.account.get",
        "parameters": [{
          "name": "teamName",
          "in": "path",
          "description": "The team name.",
          "required": true,
          "style": "simple",
          "explode": false,
          "schema": {
            "type": "string"
          }
        }, {
          "name": "environment",
          "in": "path",
          "description": "The account environment.",
          "required": true,
          "style": "simple",
          "explode": false,
          "schema": {
            "type": "string"
          }
        }],
        "responses": {
          "200": {
            "description": "An Account",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/LinkedAccount"
                }
              }
            }
          },
          "404": {
            "description": "Resource was not found",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/BaseError"
                }
              }
            }
          }
        }
      },
      "patch": {
        "summary": "Update Team Account",
        "operationId": "api.account.update",
        "parameters": [{
          "name": "teamName",
          "in": "path",
          "description": "The team name.",
          "required": true,
          "style": "simple",
          "explode": false,
          "schema": {
            "type": "string"
          }
        }, {
          "name": "environment",
          "in": "path",
          "description": "The account environment.",
          "required": true,
          "style": "simple",
          "explode": false,
          "schema": {
            "type": "string"
          }
        }],
        "requestBody": {
          "$ref": "#/components/requestBodies/AccountModify"
        },
        "responses": {
          "200": {
            "description": "An Account",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/LinkedAccount"
                }
              }
            }
          },
          "404": {
            "description": "Resource was not found",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/BaseError"
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "BaseError": {
        "required": [
          "code",
          "message"
        ],
        "type": "object",
        "properties": {
          "message": {
            "type": "string"
          },
          "code": {
            "type": "integer"
          }
        }
      },
      "BaseStringList": {
        "type": "array",
        "items": {
          "type": "string"
        }
      },
      "BaseLink": {
        "type": "object",
        "properties": {
          "rel": {
            "type": "string"
          },
          "href": {
            "type": "string",
            "format": "uri"
          }
        }
      },
      "BaseLinkList": {
        "type": "object",
        "properties": {
          "links": {
            "type": "array",
            "items": {
              "$ref": "#/components/schemas/BaseLink"
            }
          }
        }
      },
      "BaseTeam": {
        "type": "object",
        "properties": {
          "teamName": {
            "type": "string"
          }
        }
      },
      "LinkedTeam": {
        "allOf": [{
          "$ref": "#/components/schemas/BaseTeam"
        }, {
          "$ref": "#/components/schemas/BaseLinkList"
        }]
      },
      "BaseTeamList": {
        "type": "object",
        "properties": {
          "data": {
            "type": "array",
            "items": {
              "$ref": "#/components/schemas/LinkedTeam"
            }
          }
        }
      },
      "LinkedTeamList": {
        "allOf": [{
          "$ref": "#/components/schemas/BaseTeamList"
        }, {
          "$ref": "#/components/schemas/BaseLinkList"
        }]
      },
      "BaseADGroups": {
        "type": "object",
        "properties": {
          "deploy": {
            "$ref": "#/components/schemas/BaseStringList"
          }
        }
      },
      "BaseVPC": {
        "type": "object",
        "properties": {
          "cidr": {
            "type": "string"
          }
        }
      },
      "BaseSubnet": {
        "type": "object",
        "properties": {
          "vpc": {
            "$ref": "#/components/schemas/BaseVPC"
          }
        }
      },
      "BaseRegion": {
        "type": "object",
        "properties": {
          "regionName": {
            "type": "string"
          },
          "subnet": {
            "$ref": "#/components/schemas/BaseSubnet"
          }
        }
      },
      "BaseRegionList": {
        "type": "array",
        "items": {
          "$ref": "#/components/schemas/BaseRegion"
        }
      },
      "BaseAccountModifiable": {
        "type": "object",
        "properties": {
          "adGroups": {
            "$ref": "#/components/schemas/BaseADGroups"
          }
        }
      },
      "BaseAccountCreate": {
        "allOf": [{
          "$ref": "#/components/schemas/BaseAccountModifiable"
        }, {
          "type": "object",
          "properties": {
            "organizationName": {
              "type": "string"
            },
            "teamName": {
              "maxLength": 12,
              "minLength": 1,
              "type": "string"
            }
          }
        }]
      },
      "BaseAccount": {
        "allOf": [{
          "$ref": "#/components/schemas/BaseAccountCreate"
        }, {
          "type": "object",
          "properties": {
            "regionMapping": {
              "$ref": "#/components/schemas/BaseRegionList"
            },
            "accountName": {
              "type": "string"
            },
            "accountStatus": {
              "type": "string"
            }
          }
        }]
      },
      "LinkedAccount": {
        "allOf": [{
          "$ref": "#/components/schemas/BaseAccount"
        }, {
          "$ref": "#/components/schemas/BaseLinkList"
        }]
      },
      "BaseAccountList": {
        "type": "object",
        "properties": {
          "data": {
            "type": "array",
            "items": {
              "$ref": "#/components/schemas/LinkedAccount"
            }
          }
        }
      },
      "LinkedAccountList": {
        "allOf": [{
          "$ref": "#/components/schemas/BaseAccountList"
        }, {
          "$ref": "#/components/schemas/BaseLinkList"
        }]
      }
    },
    "responses": {
      "NotFound": {
        "description": "Resource was not found",
        "content": {
          "application/json": {
            "schema": {
              "$ref": "#/components/schemas/BaseError"
            }
          }
        }
      },
      "Root": {
        "description": "List of Routes",
        "content": {
          "application/json": {
            "schema": {
              "$ref": "#/components/schemas/BaseLinkList"
            }
          }
        }
      },
      "TeamList": {
        "description": "List of Teams",
        "content": {
          "application/json": {
            "schema": {
              "$ref": "#/components/schemas/LinkedTeamList"
            }
          }
        }
      },
      "AccountList": {
        "description": "List of Accounts",
        "content": {
          "application/json": {
            "schema": {
              "$ref": "#/components/schemas/LinkedAccountList"
            }
          }
        }
      },
      "Account": {
        "description": "An Account",
        "content": {
          "application/json": {
            "schema": {
              "$ref": "#/components/schemas/LinkedAccount"
            }
          }
        }
      }
    },
    "parameters": {
      "TeamName": {
        "name": "teamName",
        "in": "path",
        "description": "The team name.",
        "required": true,
        "style": "simple",
        "explode": false,
        "schema": {
          "type": "string"
        }
      },
      "Environment": {
        "name": "environment",
        "in": "path",
        "description": "The account environment.",
        "required": true,
        "style": "simple",
        "explode": false,
        "schema": {
          "type": "string"
        }
      }
    },
    "requestBodies": {
      "AccountCreate": {
        "description": "Account Update JSON",
        "content": {
          "application/json": {
            "schema": {
              "$ref": "#/components/schemas/BaseAccountCreate"
            }
          }
        },
        "required": true
      },
      "AccountModify": {
        "description": "Account Create JSON",
        "content": {
          "application/json": {
            "schema": {
              "$ref": "#/components/schemas/BaseAccountModifiable"
            }
          }
        },
        "required": true
      }
    }
  }
}
```

</p>
</details>

### Describe the bug you're encountering
When `components` comes before `paths` in YAML/JSON specification Schemas are not rendered correctly.

### To reproduce...

1. Copy the `Working` JSON to https://editor.swagger.io/
2. Expand `[GET] /teams/{teamName}/{environment}` route.
3. Examine the Code 200 Example Response or Schema.
4. Copy the `Broken` JSON to https://editor.swagger.io/
5. Expand `[GET] /teams/{teamName}/{environment}` route.
6. Examine the Code 200 Example Response or Schema.

### Expected behavior
The following attributes should exist in the Schema:
* regionMapping
* accountName
* accountStatus
* links

### Additional context or thoughts
I'm using https://github.com/zalando/connexion, when it converts the YAML spec to JSON order is not preserved and `components` ends up coming before `paths` breaking the Schema rendering.

Also, if the Schema is rendered inside an array then all the attributes show up regardless of where `components` is defined.

**Comment by tahoward** [comment]
Temporary solution is to pass JSON_SORT_KEYS = False to Flask app to preserve key order. Would assume expectation is key order should not matter to swagger-ui?

**Comment by shockey** [comment]
> Would assume expectation is key order should not matter to swagger-ui?

In a perfect world, yes! Our resolver engine has some corner cases around this, though. This is a bug!

**Comment by augensalat** [comment]
May I add this sparse example?
```yaml
openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
servers:
  - url: /
    
components:
  schemas:
    Pet:
      allOf:
        - $ref: '#/components/schemas/createPet'
      required:
        - id
      properties:
        id:
          type: integer
    createPet:
      required:
        - name
      properties:
        name:
          type: string
        tag:
          type: string
    Pets:
      type: array
      items:
        $ref: "#/components/schemas/Pet"
  responses:
    listPets:
      description: An array of pets
      content:
        application/json:    
          schema:
            $ref: "#/components/schemas/Pets"
    getPet:
      description: An array of pets
      content:
        application/json:    
          schema:
            $ref: "#/components/schemas/Pet"

paths:
  /pets:
    get:
      summary: List all pets
      responses:
        '200':
          "$ref": "#/components/responses/listPets"
  /pets/{petId}:
    get:
      summary: Info for a specific pet
      tags:
        - pets
      parameters:
        - name: petId
          in: path
          required: true
          description: The id of the pet to retrieve
          schema:
            type: string
      responses:
        '200':
          "$ref": "#/components/responses/getPet"
```
`GET /pets/{petId}` correctly returns `#/components/schemas/Pet` with all (three) elements.
However in `GET /pets`, that just returns an array of the same `#/components/schemas/Pet`, only the `id` is shown in the array objects.

As already described, a workaround is moving `paths` before `components`, but as for @tahoward this is no option in my case.

**Comment by shockey** [comment]
Looks like this has at lease some overlap with #4672.

**Comment by tim-lai** [comment]
fixed in [swagger-ui@3.36.1](https://github.com/swagger-api/swagger-ui/releases/tag/v3.36.1)
